src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.climateConditions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.AliveCharacterComponent;
+import org.terasology.logic.chat.ChatMessageEvent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.registry.In;
+import org.terasology.world.WorldComponent;
+
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class BodyTemperatureSystem extends BaseComponentSystem {
+    public static final String BODY_TEMPERATURE_UPDATE_ACTION_ID = "Body Temperature Update";
+    public static final String BODY_TEMPERATURE_DISPLAY_ACTION_ID = "Body Temperature Display";
+
+    private static final Logger logger = LoggerFactory.getLogger(BodyTemperatureSystem.class);
+
+    @In
+    private EntityManager entityManager;
+    @In
+    ClimateConditionsSystem climateConditionsSystem;
+    @In
+    DelayManager delayManager;
+
+    private static final int CHECK_INTERVAL = 1000;
+    private static final int DISPLAY_INTERVAL = 10000;
+
+    public void postBegin() {
+        boolean processedOnce = false;
+        for (EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)) {
```

Comment:
why get entities with `WorldComponent` :thinking: not sure I understand the intention here :sweat_smile:

Reply:
I'm not sure how multiplayer works, but here, if I check all the player entities.. then there's a chance every time a new player joins a server, a new body update periodic action will be added which would act on all player entities. Since, we need to add the periodic action in postBegin(), the only way to make it act on the player entity is to use entity manager to get all entities with CharacterAliveComponent and some other components which would not single out the player alone it would get all similar entities (i.e. all other players on the server).
I figured this must be true for other systems so I checked how they tackle it, and then the thirstAuthoritySystem did this, so I figured the WorldComponent must mean the world and hence I did what I did.

Reply:
I see, tbh I forgot that the delayManager expects an entity on which to trigger the periodic action.
It's absolutely correct to not do this on any player but on the world instead :+1:
I currently don't know a better way to do this, so it's fine to do it like the thirst authority system.